resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceTypeHelper.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.yarn
+
+import java.lang.reflect.InvocationTargetException
+
+import scala.util.control.NonFatal
+
+import org.apache.hadoop.yarn.api.records.Resource
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.Utils
+
+private[yarn] class ResourceTypeHelper {
+}
+
+object ResourceTypeHelper extends Logging {
+  private val resourceTypesNotAvailableErrorMessage =
+    "Ignoring updating resource with resource types because " +
+    "the version of YARN does not support it!"
+
+  def setResourceInfoFromResourceTypes(resourceTypesParam: Map[String, String],
+                                       resource: Resource): Resource = {
+    if (resource == null) {
+      throw new IllegalArgumentException("Resource parameter should not be null!")
+    }
+
+    if (!ResourceTypeHelper.isYarnResourceTypesAvailable()) {
+      logWarning(resourceTypesNotAvailableErrorMessage)
+      return resource
+    }
+
+    val resourceTypes = resourceTypesParam.map { case (k, v) => (
+      if (k.equals("memory")) {
+        logWarning("Trying to use 'memory' as a custom resource, converted it to 'memory-mb'")
+        "memory-mb"
+      } else k, v)
+    }
+
+    logDebug(s"Size of custom resource types: ${resourceTypes.size}")
```

Comment:
Not really a helpful debug log.

Reply:
Modified it to be more helpful, so the whole resource map is logged.